types/events.go

```diff
@@ -280,6 +295,10 @@ type BlockEventPublisher interface {
 	PublishEventValidatorSetUpdates(context.Context, EventDataValidatorSetUpdates) error
 }
 
+//Todo EvidenceEventPublisher
+type EvidenceEventPublisher interface {
```

Comment:
Do we need this interface? (I'm not sure why we need the one below it either, but that one was pre-existing)

Reply:
Right now both EventBus and NopEventBus implement all these interfaces (BlockEventPublisher, EvidenceEventPublisher and TxEventPublisher). I think the initial idea might have been that this was a clean way to add functions into one of the groups and enforce implementation of them. Since we have only one function in the two interfaces we might just add these functions, but as we have two structs with the same functions, I think its ok to keep the interface.

Reply:
> Right now both EventBus and NopEventBus implement all these interfaces (BlockEventPublisher, EvidenceEventPublisher and TxEventPublisher). 

Right, but what I meant was we don't seem to use this anywhere. The BlockEventPublisher is used in one place, but everywhere else (as far as I can see) uses the real event bus. 

As a general convention it's better to only define interfaces where they're needed (consumed) rather than exporting them from very central library packages where they're not implemented. I know we have existing patterns that did so, but I'm hoping to clean that up and if we can avoid extending it I'd like to. 

If some package only cares about a subset, I think it's preferable to define that subset interface there (where it's used) rather than in the base.

Reply:
I agree with your comment. Actually I just realized that the function in this interface is implemented only in EventBus (I mixed it up with PublishEventNewEvidence which is implemented in both.

So we can entirely remove it from here. Now, if we ever plan to add new events related to uncommitted evidence then it seems cleaner to me to have an interface (defined in event_bus.go where EventBus is), than just functions. 

So would you suggest we just move the interface or remove it and keep the function?

Reply:
> So would you suggest we just move the interface or remove it and keep the function?

Personally I'd just remove the interface and keep the function. The interface doesn't have to be predefined if we aren't using it.